### PR TITLE
block_devices_plug: Fix support pci bus specified

### PR DESCRIPTION
**Pull request by YongxueHong**
Support the pci bus specified on creating block devices.

ID: 1846973
Signed-off-by: Yongxue Hong <yhong@redhat.com>

### CODE REVIEW

File: provider/block_devices_plug.py [hunk 1]
```diff
@@ -234,7 +234,7 @@ def _create_devices(self, images):
             img_params = self.vm.params.object_params(img)
             devices_created = getattr(
                 self.vm.devices, '%s_define_by_params' % self._dev_type['name'])(
-                img, img_params, self._dev_type['media'])
+                img, img_params, self._dev_type['media'], pci_bus=pci_bus)
```
Comment (vivianQizhu) [inline provider/block_devices_plug.py:237]: I am afraid this would fail those device generator that has no parameter `pci_bus`

Comment (YongxueHong) [reply provider/block_devices_plug.py:237]: Hi, @vivianQizhu 
This module just care about the block devices which have the parameter `pci_bus`.
like:
```
    def cdroms_define_by_params(self, name, image_params, media=None,
                                index=None, image_boot=None,
                                image_bootindex=None,
                                pci_bus={"aobject": "pci.0"}):
```
```
    def images_define_by_params(self, name, image_params, media=None,
                                index=None, image_boot=None,
                                image_bootindex=None,
                                pci_bus={"aobject": "pci.0"}):

```
Thanks.

Comment (vivianQizhu) [reply provider/block_devices_plug.py:237]: @YongxueHong Ah okay, then it would be fine.

**Comment by vivianQizhu** [comment]
Test passed:
02:30:28 DEBUG| Send command: {'execute': 'device_add', 'arguments': OrderedDict([('driver', 'virtio-blk-pci'), ('id', 'stg8'), ('drive', 'drive_stg8'), ('write-cache', 'on'), ('multifunction', 'on'), ('addr', '0x0.0x0'), ('bus', 'pcie_extra_root_port_0')]), 'id': 'lVLiTZU8'}

**Comment by vivianQizhu** [comment]
I am merging it, let's adjust it later if hit other issues, thanks @YongxueHong

**Comment by YongxueHong** [comment]
> I am merging it, let's adjust it later if hit other issues, thanks @YongxueHong

Okay, I agree it.

**Pull request merged**